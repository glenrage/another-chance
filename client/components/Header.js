import React from 'react';
import { Link } from 'react-router';

const LoggedOutView = (props) => {
  if (!props.currentUser) {
    return (
      <ul className="nav navbar-nav navbar-right">
        <li className="nav-item">
          <Link to="/" className="nav-link">
            <i className="fa fa-paw" />
            Casa
          </Link>
        </li>

        <li className="nav-item">
          <Link to="login" className="nav-link">
            <i className="fa fa-sign-in" aria-hidden="true" />
            Iniciar Sesi√≥n
          </Link>
        </li>

        <li className="nav-item">
          <Link to="register" className="nav-link">
            <i className="fa fa-user-plus" aria-hidden="true" />
          Registrar
          </Link>
        </li>
      </ul>
    );
  }
  return null;
};

const LoggedInView = (props) => {
  if (props.currentUser) {
    return (
      <ul className="nav navbar-nav navbar-right">

        <li className="nav-item">
          <Link to="/" className="nav-link">
            <i className="fa fa-paw" />
            Casa
          </Link>
        </li>

        <li className="nav-item">
          <Link to="animalform" className="nav-link">
            <i className="fa fa-plus" />Nuevo Animal
          </Link>
        </li>

        <li className="nav-item">
          <Link to="animals" className="nav-link">
            <i className="fa fa-search" />Animals
          </Link>
        </li>

        <li className="nav-item">
          <Link
            to="settings"
            className="nav-link"
          >
            <i className="fa fa-user-circle-o" aria-hidden="true" />
            {props.currentUser.firstName}
          </Link>
        </li>
      </ul>
    );
  }

  return null;
};


class Header extends React.Component {
  render() {
    return (

      <nav className="navbar navbar-full">
        <Link to="/" className="navbar-brand">
          <img id="nav-img" src="/assets/images/logo.jpg" alt="Logo" />
        </Link>

        <LoggedOutView currentUser={this.props.currentUser} />

        <LoggedInView currentUser={this.props.currentUser} />


      </nav>


    );
  }
}

export default Header;
