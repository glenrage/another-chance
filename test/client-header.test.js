import React from 'react';
import Header from '../client/components/Header';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('Header snapshot test', () => {
  const component = shallow(<Header />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
