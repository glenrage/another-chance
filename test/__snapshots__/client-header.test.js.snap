// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test 1`] = `
<nav
  className="navbar navbar-full"
>
  <Link
    className="navbar-brand"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/"
  >
    <i
      className="fa fa-heartbeat"
    />
    Colitas Por La Vida
  </Link>
  <LoggedOutView />
  <LoggedInView />
</nav>
`;
